### PR TITLE
Blog - An Introduction to Kubernetes Secrets

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by balapriyac** [comment]
Adding the preview link: https://63ea968da9556f27ffc8ba9d--ephemeral-narwhal-3ac3b2.netlify.app/

**Comment by balapriyac** [comment]
@msali123 can you add an  image or two?

**Comment by msali123** [comment]
Preview URL:
https://63eb49db2097765e4b1a0a6a--ephemeral-narwhal-3ac3b2.netlify.app

File: blog/_posts/2023-03-17-kubernetes-secrets.md [hunk 1]
```diff
@@ -0,0 +1,671 @@
...
+![Viewing pod private-pod]({{site.images}}{{page.slug}}/Zd61bEv.png)
+</div>
+
+## Conclusion
+
+In this tutorial, you have learned about Kubernetes secrets. You have also learned how to use a secret as an environmental variable and as a volume mount. In addition, you know how to create a secret from a file, from the command line using `kubectl` , and from YAML manifest files.
+
+You then learned how to go through the process of authenticating your Kubernetes cluster to pull an image from your private repository on DockerHub by creating a secret from the `.docker/config.json` file, and pulling the image using the `imagePullSecrets` attribute.
+
+Now that you have a good knowledge of Kubernetes secrets, you can now implement it in your application and deploy safely to Kubernetes.
```
Comment (balapriyac) [inline blog/_posts/2023-03-17-kubernetes-secrets.md:669]:
```suggestion
Now that you have a good knowledge of Kubernetes secrets, you can implement it in your application and deploy safely to Kubernetes.
```

File: blog/_posts/2023-03-17-kubernetes-secrets.md [hunk 2]
```diff
@@ -0,0 +1,671 @@
...
+kubectl get secret -n example
+~~~
+
+You can see the secret `postgres-demo` was created with a secret type of **opaque:**
+
+<div class="wide">
+![Viewing Secret postgres-demo]({{site.images}}{{page.slug}}/83fPibh.png)
+</div>
+
+In Kubernetes there are various types of secrets, we have:
```
Comment (balapriyac) [inline blog/_posts/2023-03-17-kubernetes-secrets.md:86]:
```suggestion
In Kubernetes there are various types of secrets; we have:
```

File: blog/_posts/2023-03-17-kubernetes-secrets.md [hunk 3]
```diff
@@ -0,0 +1,671 @@
...
+![Creating Secret(postgres-demo-0)]({{site.images}}{{page.slug}}/7GB0UUb.png)
+</div>
+
+Run the command below, to see the newly created secret:
+
+~~~{.bash caption=">_"}
+kubectl get secret -n example
+~~~
+
+You should now have two secrets - *postgres-demo*  and *postgres-demo-0*, as shown below:
```
Comment (balapriyac) [inline blog/_posts/2023-03-17-kubernetes-secrets.md:140]:
```suggestion
You should now have two secrets: *postgres-demo*  and *postgres-demo-0*, as shown below:
```

File: blog/_posts/2023-03-17-kubernetes-secrets.md [hunk 4]
```diff
@@ -0,0 +1,671 @@
...
+
+<div class="wide">
+![Verifying Persistent Volume and Claim]({{site.images}}{{page.slug}}/7LGtteC.png)\
+</div>
+
+### Using a Kubernetes Secret as an Environment Variable
+
+Up to this point you have created a persistent volume (PV) and a persistent volume claim (PVC), you are now ready to configure a [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) and a service to deploy a PostgreSQL database up to Kubernetes.
+
+The secret **postgres-secret* holds the credentials needed to access the PostgreSQL server, so you will use that secret as an environment variable to configure a statefulSet to deploy a PostgreSQL database.
```
Comment (balapriyac) [inline blog/_posts/2023-03-17-kubernetes-secrets.md:273]:
```suggestion
The secret **postgres-secret** holds the credentials needed to access the PostgreSQL server, so you will use that secret as an environment variable to configure a statefulSet to deploy a PostgreSQL database.
```

File: blog/_posts/2023-03-17-kubernetes-secrets.md [hunk 5]
```diff
@@ -0,0 +1,671 @@
...
+
+Here's what we are doing in the code above:
+
+Define a volume section under the statefulSet Pod specification, with a name *secret-volume*.
+Specify the type of volume (a secret volume type) and which secret it should use (*postgres-secret*).
+Include the keys Kubernetes should watch out for while creating the statefulset (*postgres_password* and *postgres_username*) alongside their paths which are required *postgres_password* and *postgres_username* (these paths could be anything, like *my-postgres-path*).
+Define a mode **511** which is optional.
+Mount the volume inside the containers section using *volumeMounts* , specify the name of the volume we'd like to mount which is *secret-volume* and then the *mountPath* `/secret` which is where the secret will be saved.
+Added a `env` section to get the POSTGRES_USER and POSTGRES_PASSWORD value form the  */var/lib/postgresql/secret/postgres_username* and */var/lib/postgresql/secret/postgres_password* file paths in the container's file system.
+
```
Comment (balapriyac) [inline blog/_posts/2023-03-17-kubernetes-secrets.md:474]:
```suggestion
- Define a volume section under the statefulSet Pod specification, with a name *secret-volume*.
- Specify the type of volume (a secret volume type) and which secret it should use (*postgres-secret*).
- Include the keys Kubernetes should watch out for while creating the statefulset (*postgres_password* and *postgres_username*) alongside their paths which are required (these paths could be anything, like *my-postgres-path*).
- Define a mode **511** which is optional.
- Mount the volume inside the containers section using *volumeMounts* , specify the name of the volume we'd like to mount which is *secret-volume* and then the *mountPath* `/secret` which is where the secret will be saved.
- Add an `env` section to get the POSTGRES_USER and POSTGRES_PASSWORD value from the  */var/lib/postgresql/secret/postgres_username* and */var/lib/postgresql/secret/postgres_password* file paths in the container's file system.

```

**Review by balapriyac — CHANGES_REQUESTED** [review]
Looks great, thanks Sameer. @jalletto I've finished reviewing this PR.

**Comment by msali123** [comment]
@balapriyac done with changes.

**Comment by msali123** [comment]
Preview URL:
https://63ec95ea0c4f967daa4a6a7e--ephemeral-narwhal-3ac3b2.netlify.app

**Pull request merged**